packages/block-editor/src/components/block-list/content.scss

```diff
@@ -183,6 +183,12 @@ _::-webkit-full-page-media, _:future, :root .has-multi-selection .block-editor-b
 		user-select: none;
 	}
 
+	&.is-selected,
+	&.has-child-selected {
+		// Bring the selected block forward so we can see it.
+		z-index: z-index(".block-editor-block-list__block.is-selected");
+	}
+
```

Comment:
One potential subtle issue with setting the z-index to `1` is for sticky Group blocks, the z-index will be set to `10`, so it's possible for this to result in the current z-index when the block is selected being lower than expected. It's pretty subtle, so I'm not sure how worth it is to try to workaround it, but here's an example:

Take a page that has a number of sticky Group blocks wrapping heading blocks that are meant to stick to the top as you scroll the page. Each subsequent sticky heading should overlap the previous one in this case. If I have no blocks selected, this works fine. Then, if I go to select the third sticky Group block, with it selected, the second sticky Group block appears to sit over the top of the 3rd one, because while selected the 3rd one has a z-index of `1`:

https://github.com/WordPress/gutenberg/assets/14988353/e24fa8e9-1272-422d-8848-fd6f3cb4a1bd

Again, I'm not sure if this is would be a blocker, but a couple of potential ideas:

* Try increasing the z-index for selected blocks so that it's at least the same z-index as sticky blocks. And/or:
* Only apply this z-index rule when negative margins are present. For that, one approach could be to output a classname to indicate (in the block editor only) if a negative margin is attached to a block?

Aside from that, this feature is testing quite well for me. Great work so far!

Reply:
Any reason we can't set the z-index on select to a ludicrously high number? A would that solve it? Bringing a group to the foreground in the editor on select where the contents are output in a DOM (as compared to canvas apps like Figma where the order of the markup doesn't matter) makes sense to me to have applied in a generic way. At least to start with, and then we can revisit if need be.

Reply:
I didn't like the idea of a 9999, but I might have been to conservative with the 1 (I went with the lowest value that would work). We can set it to 20?

Reply:
I think the idea to alter the z-index only if negative margins are set has some merit. This way we're sure not to interfere with other things except when there is a reason.

Reply:
I would avoid that simply because it's code complexity built just for this one thing, and we don't even know if it's necessary. On the flipside we _do_ have a fairly good idea that bringing to the front on select can benefit other contexts, like editing sliders or absolute positioned elements, which we'll get at some point.

Reply:
The problem is it's non-trivial to limit this, no? And we don't even know that it's needed. I'd almost rather _not_ have an editor affordance to bring it to the front on select, than ship a hack attached only to when you were using negative margins.

Reply:
I added the class [here](https://github.com/WordPress/gutenberg/pull/60347/commits/403a85a4dd9725fbad2653918f8fa76977691c38). I don't know how terrible it is technically as a solution? I can revert if we don't want it in the end

Reply:
I'd avoid that class too, especially if it's output also on the frontend. It's not a class that's easy to remove again once added. I'd plead to please consider it separate from this PR entirely, it seems like the wrong path forward.

Reply:
> I'd avoid that class too, especially if it's output also on the frontend. It's not a class that's easy to remove again once added. I'd plead to please consider it separate from this PR entirely, it seems like the wrong path forward.

no, it's editor ONLY, we have a few of those, just like is-selected is one of them

Reply:
Okay! I'll defer to you! Thanks for the work.

Reply:
The editor only classname of `.has-negative-margin` combined with the `z-index` of `20` sounds like a really good workaround to me! (z-index is now higher than sticky but still lower than other UI elements, toolbars, etc). Nice work untangling this @MaggieCabrera 👍